app/models/user.rb

```diff
@@ -1,6 +1,25 @@
 class User < ApplicationRecord
-  # Include default devise modules. Others available are:
-  # :lockable, :timeoutable and :omniauthable
   devise :database_authenticatable, :registerable,
-         :recoverable, :rememberable, :trackable, :validatable, :confirmable
+         :recoverable, :rememberable, :trackable, :validatable,
+         :confirmable, :omniauthable, omniauth_providers: [:github, :google_oauth2]
+
+  def self.from_omniauth(auth)
+    where(provider: auth.provider, uid: auth.uid).first_or_create do |user|
+      user.email = auth.info.email
+      user.password = Devise.friendly_token[0,20]
+      # user.name = auth.info.name
+      # user.image = auth.info.image
+      user.skip_confirmation!
+    end
+  end
+
+  def self.new_with_session(params, session)
+    super.tap do |user|
+      %w[devise google_oauth2].each do |provider|
+        if data = session["#{provider}.github_data"] && session["#{provider}.github_data"]["extra"]["raw_info"]
```

Comment:
That seems a bit hacky to me. Could we remove this part? And what the consequences?

Reply:
hmm... that is right from the docs :
https://github.com/zquestz/omniauth-google-oauth2#devise

The same story was with facebook oauth gem. For github - there was no info in official docs on it, but if also found it somewhere in manual